client/components/data/query-inline-help-search/index.jsx

```diff
@@ -0,0 +1,49 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestInlineHelpSearchResults } from 'state/inline-help/actions';
+import { isRequestingInlineHelpSearchResultsForQuery } from 'state/inline-help/selectors';
+
+class QueryInlineHelpSearch extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.query === nextProps.query ) {
+			return;
+		}
+
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( props.requesting || ! props.query ) {
+			return;
+		}
+
+		props.requestInlineHelpSearchResults( props.query );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	( state, ownProps ) => {
```

Comment:
`( state, { query } ) => {` might save you the `const` part and would allow this to be simplified to just pure object return like I've mentioned in other comment above

Reply:
Nice, good call! 👍